astrodata/fits.py

```diff
@@ -1039,6 +1166,9 @@ def to_hdulist(self):
             if ext.mask is not None:
                 hlst.append(new_imagehdu(ext.mask, header, 'DQ'))
 
+            if isinstance(ext.wcs, WCS):
+                hlst.append(wcs_to_asdftablehdu(ext.wcs, extver=ver))
```

Comment:
What happens if `wcs` is an astropy `WCS` object, not a `gwcs` one ?
For consistency and more general use, would it be possible to fallback on loading the astropy `WCS` from the header if the `(g)wcs` extension is missing?

Reply:
Yes, @csimpson and I have discussed that and just haven't got to it yet. In the meantime, this adds functionality that doesn't already exist... My main purpose here was to demonstrate that we can serialize a `gwcs` attribute and get the use case for spectroscopy working.

Reply:
The idea is additionally to keep a FITS approximation of the gWCS solution so that eg. display tools can show useful co-ordinates, but that obviously requires more work. I've also discussed this a bit with Nadia recently and they are planning to use a look-up table for that purpose, which (although highly redundant) makes some sense, since it doesn't actually need to be an approximation or difficult to generate.